rust/arrow/src/array/array.rs

```diff
@@ -1038,7 +1159,48 @@ impl From<ArrayDataRef> for BinaryArray {
     }
 }
 
-impl<'a> From<Vec<&'a str>> for BinaryArray {
+impl From<ArrayDataRef> for StringArray {
+    fn from(data: ArrayDataRef) -> Self {
+        assert_eq!(
+            data.buffers().len(),
+            2,
+            "StringArray data should contain 2 buffers only (offsets and values)"
+        );
+        let raw_value_offsets = data.buffers()[0].raw_data();
+        assert!(
```

Comment:
This may be unnecessary because Buffer already has alignment check

Reply:
I removed it and got test failures, I've left it in for now